pulp_rpm/app/depsolving.py

```diff
@@ -135,14 +133,16 @@ def rpm_filelist_conversion(solvable, unit):
         repodata = solv_repo.first_repodata()
 
         for file_repr in unit.get('files', []):
-            dir_path = file_repr[1]
-            if not dir_path:
+            # file_repr = e.g. (None, '/usr/bin/', 'bash')
+            file_dir = file_repr[1]
+            file_name = file_repr[2]
+            if not file_dir:
                 # https://github.com/openSUSE/libsolv/issues/397
                 continue
-            dirname_id = repodata.str2dir(dir_path)
+            dirname_id = repodata.str2dir(file_dir)
             repodata.add_dirstr(
                 solvable.id, solv.SOLVABLE_FILELIST,
-                dirname_id, os.path.basename(dir_path)
```

Comment:
The problem: we were never adding the actual filenames, just the "basename" of the directory they're in, which is nonsensical.

Reply:
uh oh

Reply:
Great catch! Tough to narrow down